src/battle_script_commands.c

```diff
@@ -5537,6 +5537,16 @@ static void Cmd_moveend(void)
             }
             gBattleScripting.moveendState++;
             break;
+        case MOVEEND_NUM_HITS:
+            if (gBattlerAttacker != gBattlerTarget
```

Comment:
Missing `TARGET_TURN_DAMAGED` and `gBattleMons[gBattlerTarget].hp != 0`

Reply:
> `TARGET_TURN_DAMAGED`

As far as I understand it you don't need to take damage as long as you get hit by a damage dealing move (e.g. False Swipes).

> `gBattleMons[gBattlerTarget].hp != 0`

What is the reason for this check?

Reply:
> As far as I understand it you don't need to take damage as long as you get hit by a damage dealing move (e.g. False Swipes).
`TARGET_TURN_DAMAGED` should still evaluate to true for False Swipe. If it doesn't that's a bug. You need it because otherwise this branch will be triggered by Pain Split (by virtue of the power being set to 1) even though it doesn't do direct damage.

> What is the reason for this check?
Looked it up, and you actually shouldn't add it in the end. The counter's not reset upon fainting like with othercounters.

Reply:
> `TARGET_TURN_DAMAGED` should still evaluate to true for False Swipe. If it doesn't that's a bug. You need it because otherwise this branch will be triggered by Pain Split (by virtue of the power being set to 1) even though it doesn't do direct damage.

Not sure what to do about `TARGET_TURN_DAMAGED`. It checks for non zero damage so I would need a workaround for False Swipe, Endure and Focus Band since all of those can result in 0 damage, right?

Reply:
You should double check if it's actually not evaluating to true for False Swipe. Because if it is, certain branches of code like:
```
        case ABILITY_STATIC:
            if (!(gMoveResultFlags & MOVE_RESULT_NO_EFFECT)
             && gBattleMons[gBattlerAttacker].hp != 0
             && !gProtectStructs[gBattlerAttacker].confusionSelfDmg
             && TARGET_TURN_DAMAGED
             && CanBeParalyzed(gBattlerAttacker)
             && IsMoveMakingContact(move, gBattlerAttacker)
             && RandomWeighted(RNG_STATIC, 2, 1))
```

Will not be triggered in those cases you mentioned even though they very much should be.

Reply:
And if it is indeed not, look at how setting those values is handled in the CFRU. They're always set to at least `1` if the target is damaged.

Reply:
Yep, Static and False Swipe interaction is also bugged. 

As far as I can tell `gHpDealt` is set to `gBattleMoveDamage` and I don't see where `physicalDmg` could be set to 1 if False Swipes deals 0 damage. 
```
if (gBattleMons[battler].hp > gBattleMoveDamage)
{
    gBattleMons[battler].hp -= gBattleMoveDamage;
    gHpDealt = gBattleMoveDamage;
}
``` 
I looked into CFRU and unless I'm misunderstand it the only time `hpDealt` can be set to 1 is if Raid Shields are up
```
if (HasRaidShields(gActiveBattler)) //Raid Shields can be used outside of Raid Battles now
    hpDealt = MathMax(1, hpDealt); //Because damage can get heavily reduced to 0
```
Unless I'm looking at the wrong place but I don't know why `gBattleMoveDamage` would be set to 1 in this case when `Cmd_datahpupdate` is called.